[
    {
        "number": 7,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "manifest_path": "docker/linux/build_scripts/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-mq26-g339-26xf",
            "cve_id": "CVE-2023-5752",
            "summary": "Command Injection in pip when used with Mercurial",
            "description": "When installing a package from a Mercurial VCS URL, e.g. `pip install hg+...`, with pip prior to v23.3, the specified Mercurial revision could be used to inject arbitrary configuration options to the `hg clone` call (e.g. `--config`). Controlling the Mercurial configuration can modify how and which repository is installed. This vulnerability does not affect users who aren't installing from Mercurial.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-mq26-g339-26xf",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-5752",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-5752"
                },
                {
                    "url": "https://github.com/pypa/pip/pull/12306"
                },
                {
                    "url": "https://github.com/pypa/pip/commit/389cb799d0da9a840749fcd14878928467ed49b4"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pip/PYSEC-2023-228.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ"
                },
                {
                    "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E"
                },
                {
                    "url": "https://github.com/advisories/GHSA-mq26-g339-26xf"
                }
            ],
            "published_at": "2023-10-25T18:32:26Z",
            "updated_at": "2024-05-03T03:31:10Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pip"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 23.3",
                    "first_patched_version": {
                        "identifier": "23.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N",
                "score": 5.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-77",
                    "name": "Improper Neutralization of Special Elements used in a Command ('Command Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 23.3",
            "first_patched_version": {
                "identifier": "23.3"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/7",
        "html_url": "https://github.com/username/retro/security/dependabot/7",
        "created_at": "2024-05-24T01:34:44Z",
        "updated_at": "2024-05-24T01:34:44Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 6,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "setuptools"
            },
            "manifest_path": "docker/linux/build_scripts/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-r9hx-vwmv-q579",
            "cve_id": "CVE-2022-40897",
            "summary": "pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)",
            "description": "Python Packaging Authority (PyPA)'s setuptools is a library designed to facilitate packaging Python projects. Setuptools version 65.5.0 and earlier could allow remote attackers to cause a denial of service by fetching malicious HTML from a PyPI package or custom PackageIndex page due to a vulnerable Regular Expression in `package_index`. This has been patched in version 65.5.1.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-r9hx-vwmv-q579",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-40897",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-40897"
                },
                {
                    "url": "https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200"
                },
                {
                    "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
                },
                {
                    "url": "https://github.com/pypa/setuptools/issues/3659"
                },
                {
                    "url": "https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be"
                },
                {
                    "url": "https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1"
                },
                {
                    "url": "https://pyup.io/vulnerabilities/CVE-2022-40897/52495/"
                },
                {
                    "url": "https://setuptools.pypa.io/en/latest/"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20230214-0001/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-r9hx-vwmv-q579"
                }
            ],
            "published_at": "2022-12-23T00:30:23Z",
            "updated_at": "2023-05-01T13:43:54Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "setuptools"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 65.5.1",
                    "first_patched_version": {
                        "identifier": "65.5.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-1333",
                    "name": "Inefficient Regular Expression Complexity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "setuptools"
            },
            "severity": "high",
            "vulnerable_version_range": "< 65.5.1",
            "first_patched_version": {
                "identifier": "65.5.1"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/6",
        "html_url": "https://github.com/username/retro/security/dependabot/6",
        "created_at": "2024-05-24T01:34:44Z",
        "updated_at": "2024-05-24T01:34:44Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 5,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "wheel"
            },
            "manifest_path": "docker/linux/build_scripts/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-qwmp-2cf2-g9g6",
            "cve_id": "CVE-2022-40898",
            "summary": "pypa/wheel vulnerable to Regular Expression denial of service (ReDoS)",
            "description": "Python Packaging Authority (PyPA) Wheel is a reference implementation of the Python wheel packaging standard. Wheel 0.37.1 and earlier are vulnerable to a Regular Expression denial of service via attacker controlled input to the wheel cli. The vulnerable regex is used to verify the validity of Wheel file names. This has been patched in version 0.38.1.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-qwmp-2cf2-g9g6",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-40898",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-40898"
                },
                {
                    "url": "https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18"
                },
                {
                    "url": "https://pypi.org/project/wheel/"
                },
                {
                    "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
                },
                {
                    "url": "https://pyup.io/vulnerabilities/CVE-2022-40898/51499/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-qwmp-2cf2-g9g6"
                }
            ],
            "published_at": "2022-12-23T00:30:23Z",
            "updated_at": "2023-01-27T05:03:46Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "wheel"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.37.1",
                    "first_patched_version": {
                        "identifier": "0.38.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-1333",
                    "name": "Inefficient Regular Expression Complexity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "wheel"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 0.37.1",
            "first_patched_version": {
                "identifier": "0.38.1"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/5",
        "html_url": "https://github.com/username/retro/security/dependabot/5",
        "created_at": "2024-05-24T01:34:43Z",
        "updated_at": "2024-05-24T01:34:43Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 4,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "manifest_path": "docker/linux/build_scripts/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-5xp3-jfq3-5q8x",
            "cve_id": "CVE-2021-3572",
            "summary": "Improper Input Validation in pip",
            "description": "A flaw was found in python-pip in the way it handled Unicode separators in git references. A remote attacker could possibly use this issue to install a different revision on a repository. The highest threat from this vulnerability is to data integrity. This is fixed in python-pip version 21.1.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-5xp3-jfq3-5q8x",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2021-3572",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-3572"
                },
                {
                    "url": "https://github.com/pypa/pip/pull/9827"
                },
                {
                    "url": "https://github.com/pypa/pip/commit/e46bdda9711392fec0c45c1175bae6db847cb30b"
                },
                {
                    "url": "https://access.redhat.com/errata/RHSA-2021:3254"
                },
                {
                    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1962856"
                },
                {
                    "url": "https://packetstormsecurity.com/files/162712/USN-4961-1.txt"
                },
                {
                    "url": "https://www.oracle.com/security-alerts/cpuapr2022.html"
                },
                {
                    "url": "https://www.oracle.com/security-alerts/cpujul2022.html"
                },
                {
                    "url": "https://github.com/advisories/GHSA-5xp3-jfq3-5q8x"
                }
            ],
            "published_at": "2021-11-15T17:45:01Z",
            "updated_at": "2023-01-27T05:02:47Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pip"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 21.1",
                    "first_patched_version": {
                        "identifier": "21.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:N/I:H/A:N",
                "score": 5.7
            },
            "cwes": [
                {
                    "cwe_id": "CWE-20",
                    "name": "Improper Input Validation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 21.1",
            "first_patched_version": {
                "identifier": "21.1"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/4",
        "html_url": "https://github.com/username/retro/security/dependabot/4",
        "created_at": "2024-05-24T01:34:43Z",
        "updated_at": "2024-05-24T01:34:43Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 3,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "manifest_path": "docker/linux/build_scripts/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-gpvv-69j7-gwj8",
            "cve_id": "CVE-2019-20916",
            "summary": "Path Traversal in pip",
            "description": "The pip package before 19.2 for Python allows Directory Traversal when a URL is given in an install command, because a Content-Disposition header can have ../ in a filename, as demonstrated by overwriting the /root/.ssh/authorized_keys file. This occurs in _download_http_url in _internal/download.py. A fix was committed 6704f2ace.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-gpvv-69j7-gwj8",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2019-20916",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-20916"
                },
                {
                    "url": "https://github.com/pypa/pip/issues/6413"
                },
                {
                    "url": "https://github.com/gzpan123/pip/commit/a4c735b14a62f9cb864533808ac63936704f2ace"
                },
                {
                    "url": "https://github.com/pypa/pip/compare/19.1.1...19.2"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2020/09/msg00010.html"
                },
                {
                    "url": "http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00005.html"
                },
                {
                    "url": "http://lists.opensuse.org/opensuse-security-announce/2020-10/msg00010.html"
                },
                {
                    "url": "https://www.oracle.com/security-alerts/cpuapr2022.html"
                },
                {
                    "url": "https://www.oracle.com/security-alerts/cpujul2022.html"
                },
                {
                    "url": "https://github.com/advisories/GHSA-gpvv-69j7-gwj8"
                }
            ],
            "published_at": "2021-06-09T17:35:04Z",
            "updated_at": "2023-01-27T05:02:32Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pip"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 19.2",
                    "first_patched_version": {
                        "identifier": "19.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-22",
                    "name": "Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pip"
            },
            "severity": "high",
            "vulnerable_version_range": "< 19.2",
            "first_patched_version": {
                "identifier": "19.2"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/3",
        "html_url": "https://github.com/username/retro/security/dependabot/3",
        "created_at": "2024-05-24T01:34:43Z",
        "updated_at": "2024-05-24T01:34:43Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 2,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "manifest_path": "docker/linux/build_scripts/py36-requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-xqr8-7jwr-rhp7",
            "cve_id": "CVE-2023-37920",
            "summary": "Removal of e-Tugra root certificate",
            "description": "Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\n e-Tugra's root certificates are being removed pursuant to an investigation prompted by reporting of security issues in their systems. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A).",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-xqr8-7jwr-rhp7",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-37920",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7"
                },
                {
                    "url": "https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-37920"
                },
                {
                    "url": "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/certifi/PYSEC-2023-135.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-xqr8-7jwr-rhp7"
                }
            ],
            "published_at": "2023-07-25T14:43:53Z",
            "updated_at": "2023-11-09T05:05:47Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "certifi"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
                    "first_patched_version": {
                        "identifier": "2023.7.22"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-345",
                    "name": "Insufficient Verification of Data Authenticity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
            "first_patched_version": {
                "identifier": "2023.7.22"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/2",
        "html_url": "https://github.com/username/retro/security/dependabot/2",
        "created_at": "2024-05-24T01:34:43Z",
        "updated_at": "2024-05-24T01:34:43Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 1,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "manifest_path": "docker/linux/build_scripts/py36-requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-43fp-rhv2-5gv8",
            "cve_id": "CVE-2022-23491",
            "summary": "Certifi removing TrustCor root certificate",
            "description": "Certifi 2022.12.07 removes root certificates from \"TrustCor\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\nTrustCor's root certificates are being removed pursuant to an investigation prompted by media reporting that TrustCor's ownership also operated a business that produced spyware. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ).",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-43fp-rhv2-5gv8",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2022-23491",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/certifi/python-certifi/security/advisories/GHSA-43fp-rhv2-5gv8"
                },
                {
                    "url": "https://github.com/certifi/python-certifi/commit/9e9e840925d7b8e76c76fdac1fab7e6e88c1c3b8"
                },
                {
                    "url": "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-23491"
                },
                {
                    "url": "https://github.com/advisories/GHSA-43fp-rhv2-5gv8"
                }
            ],
            "published_at": "2022-12-07T23:05:18Z",
            "updated_at": "2023-01-29T05:05:37Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "certifi"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2017.11.05, < 2022.12.07",
                    "first_patched_version": {
                        "identifier": "2022.12.07"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:C/C:N/I:H/A:N",
                "score": 6.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-345",
                    "name": "Insufficient Verification of Data Authenticity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 2017.11.05, < 2022.12.07",
            "first_patched_version": {
                "identifier": "2022.12.07"
            }
        },
        "url": "https://api.github.com/repos/username/retro/dependabot/alerts/1",
        "html_url": "https://github.com/username/retro/security/dependabot/1",
        "created_at": "2024-05-24T01:34:43Z",
        "updated_at": "2024-05-24T01:34:43Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    }
]